scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/AsyncFlowDefRunner.scala

```diff
@@ -166,20 +167,27 @@ class AsyncFlowDefRunner(mode: CascadingMode) extends Writer {
       @annotation.tailrec
       def go(id: Long): Unit = messageQueue.take match {
         case Stop => ()
-        case RunFlowDef(conf, fd, promise) =>
+        case RunFlowDef(conf, fd, cpromise) =>
           try {
             val ctx = ExecutionContext.newContext(conf.setScaldingFlowCounterValue(id))(fd, mode)
             ctx.buildFlow match {
               case Success(Some(flow)) =>
                 val future = FlowListenerPromise
                   .start(flow, { f: Flow[_] => (id, JobStats(f.getFlowStats)) })
-
-                promise.completeWith(future)
+                // we want to stop the flow when the execution is cancelled
+                val cancel = new CancellationHandler {
```

Comment:
I kind of think `CancellationHandler` should be sealed... maybe this can be `CancellationHandler.fromFn(fn: () => Unit): CancelationHandler` then this can be `val cancel = CancellationHander.fromFn(() => flow.stop())`

Reply:
👍